istioctl/pkg/tag/generate_test.go

```diff
@@ -141,6 +143,14 @@ func TestGenerateValidatingWebhook(t *testing.T) {
 			name:           "webhook-pointing-to-url",
 			istioNamespace: "istio-system",
 			webhook:        revisionCanonicalWebhookRemote,
+			whURL:          remoteValidationURL,
+			whSVC:          "",
+			whCA:           "ca",
+		},
+		{
+			name:           "webhook-configured-with-bad-url",
+			istioNamespace: "istio-system",
+			webhook:        revisionCanonicalWebhookRemote,
```

Comment:
I don't understand this test. The inputs are the same as `webhook-pointing-to-url` but the expectations are different and both pass... I don't understand why we conditionally skip the test on L200

Reply:
The URL field is different in this test: `whURL:          remoteInjectionURL,`. The `webhook-pointing-to-url` test uses: `remoteValidationURL`, which is a proper validation URL. This PR is needed because the code is incorrectly using an injection URL, which this PR is correcting. As Sam's TODO on line 131 points out, this is not strictly correct, but fixing the code to do it properly is a bigger effort (so I left Sam's TODO).

Reply:
But the whURL is the _expected output_ of the test, not the inputs...? So what we are saying, in my understanding is for the same inputs, we expected 2 different outputs, and both tests pass. That only works since the test is not actually testing the expected output due to skippiung it on L200?

Reply:
You're right ... my brain isn't working so well today. I wanted to quickly test the case of a good URL, but that would require an injection webhook configured with a validation url, which never happens anyway. I think the only thing that matters is that this fix is changing injection URLs (which are always wrong), into a validation URLs. I changed the test. PTAL